Desktop integration improvements

bochecha opened the pull request:
I came about those while working on the Flatpak build (#79).

They are not required for Flatpak though, rather they make Glimpse stand apart from the upstream GNU IMP in the desktop app launcher and in software centers.

Quite a few of it is marked with TODO, simply because I didn't have the information handy. Search for those in the diff when you review it.

In addition, you might want to rephrase the app summaries/description to give it its own identity. All I did was replace all mentions of GIMP by Glimpse; that's really the minimum to avoid very confused users.

---

**Note:** This is based on the Flatpak branch for #79, do merge that one first.

igalic commented on app/gui/dbus-service.xml:
```diff
@@ -2,7 +2,7 @@
 
-  <interface name="org.gimp.GIMP.UI">
+  <interface name="app.getglimpse.Glimpse.UI">
 
```
a lot of these changes look like they are needed for the Flatpak integration as well

bochecha replied:
Yes, as I said this branch is based on the Flatpak one and so it contains the commits from the Flatpak branch. The diff you're replying to is one of those.

bochecha replied:
Now that the Flatpak branch has been merged, this branch only contains the new changes. :slightly_smiling_face:

igalic commented on build/buildbot/suppressed-warnings.txt:
```diff
@@ -18,7 +18,7 @@
 # Who cares
-:    .*gimp.desktop: warning: value .* contains .* more than once.*
+:    .*app.getglimpse.Glimpse.desktop: warning: value .* contains .* more than once.*
 :    ^libtool: install: warning:.*
```
couldn't we just fix these? rather than have them under "who cares"
also: i don't think we use buildbot, so this file can go…

bochecha replied:
That's what I asked @TrechNex on Matrix, and he said the file could be removed as well, but then I forgot and left it in. :laughing:

bochecha replied:
That was a change in the Flatpak branch, which is now merged, so marking this as resolved.

TrechNex commented:
#79 has now been merged into dev branch. This PR will need to resolve merge conflicts and then be reviewed before I can do the same here.

bochecha commented:
Rebased on the latest dev-g210.

bochecha commented on desktop/app.getglimpse.Glimpse.appdata.xml.in.in:
```diff
@@ -1,24 +1,14 @@
   <project_license>GPL-3.0+ AND LGPL-3.0+</project_license>
-  <_name>GNU Image Manipulation Program</_name>
+  <_name>Glimpse</_name>
   <_summary>Create images and edit photographs</_summary>
```
Are we happy with that summary? It's the same as the one for the GNU IMP.

BrainBlasted replied:
Looks good to me.

bochecha replied:
Ok, resolving then.

bochecha commented on desktop/app.getglimpse.Glimpse.appdata.xml.in.in:
```diff
@@ -27,418 +17,56 @@
+      procedures to be easily scripted. Glimpse is available for Linux,
       Microsoft Windows and OS X.
     </_p>
   </description>
```
Are we happy with that description? I just renamed `s/GIMP/Glimpse/` from the GNU IMP, do we want something different?

BrainBlasted replied:
For now this seems alright.

TrechNex replied:
Ditto. We can always change it later to differentiate ourselves.

bochecha commented on desktop/app.getglimpse.Glimpse.appdata.xml.in.in:
```diff
@@ -27,418 +17,56 @@
     </_p>
   </description>
+  <url type="homepage">https://getglimpse.app</url>
+  <url type="help">https://getglimpse.app/TODO</url>
```
Do we have a page on the website about getting help? If not, do we want one?

If we don't, we should remove this line.

BrainBlasted replied:
I think we'd want one eventually.

TrechNex replied:
We could link to https://wiki.getglimpse.app. Currently it's just doing project-related stuff, but it would make sense to put help pages there too if chaomodus doesn't object.

BrainBlasted replied:
I think we'd want to have a proper online manual instead of the wiki. For now maybe we should remove this or comment it out?

Member1221 replied:
Should be changed to glimpse-editor.org, no?

bochecha replied:
Since we're not ready on the server side, I commented this out. (and yes, I moved it to glimpse-editor.org)

bochecha replied:
This is now tracked in #145.

bochecha commented on desktop/app.getglimpse.Glimpse.appdata.xml.in.in:
```diff
@@ -27,418 +17,56 @@
-  <url type="donation">https://www.gimp.org/donating/</url>
-  <url type="help">https://www.gimp.org/docs/</url>
-  <url type="homepage">https://www.gimp.org/</url>
+  <update_contact>TODO@getglimpse.app</update_contact>
```
This is about who distributors should contact when the appdata file itself needs changes. It needs to be an email address, but it's optional.

Do we have such an email address? If not we should remove this line.

TrechNex replied:
I so far haven't been able to get email forwarding working properly with my domain registrar (I was the one that bought the domain name for 2 years!)

If there's a free GNU Mailman-like service where I can setup a distribution list for project maintainers please let me know, as that would probably be a good way of doing this.

bochecha replied:
Since we're not ready with this, I commented it out. (and moved it to glimpse-editor.org)

bochecha replied:
This is now tracked in #144.

bochecha commented on desktop/app.getglimpse.Glimpse.appdata.xml.in.in:
```diff
@@ -27,418 +17,56 @@
+      <image>https://getglimpse.app/TODO2.png</image>
+      <_caption>Doing more great things with Glimpse</_caption>
     </screenshot>
   </screenshots>
```
We need to add a few screenshots here, with proper captions.

They are shown in software centers (e.g GNOME Software) and help users decide whether they want to install the app.

TrechNex replied:
Good shout. The blockers for now:
* No artwork for splash screen
* My theming changes still need to be merged in (determining with community if we should be dark or light by default)

bochecha replied:
I'm not sure we should have a screenshot of the splashscreen. That doesn't seem like something users will see and go "wow, this app looks great! I should install it!".

I think we should focus on showing up the main app UI, maybe even some features in action.

BrainBlasted replied:
+1 - splash screen isn't the most important thing.

Perhaps for now these can go on wiki.getglimpse.app?

bochecha replied:
Since we're not ready yet with this, I commented them out. (and moved them to glimpse-editor.org)

bochecha replied:
This will now be tracked in #143

chaomodus commented:
Just a note on this, we need to update the URLs to *.glimpse-editor.org as general consensus has shifted to this being the main URL.

bochecha commented:
Alright, so I just rebased this and moved everything to the new glimpse-editor.org URLs.

I commented everything which was still to be done.

My suggestion is we merge it as is, and I open issues explaining what needs to be done for each of them. Is that ok?

TrechNex commented:
> My suggestion is we merge it as is, and I open issues explaining what needs to be done for each of them. Is that ok?

Sounds good to me. Just make sure you tag them against the 0.1.0 milestone

bochecha commented:
Building the Flatpak I found an issue with how I had commented the `<releases>` tag.

This is now fixed, but it invalidated the approvals. Can I get some approvals again? :slightly_smiling_face:

TrechNex commented:
Please squash this when you merge it 😂
